tests/firstTest.spec.ts

```diff
@@ -1,39 +1,53 @@
 import { test, expect } from "@playwright/test";
+import { LoginPage } from "../pages/login.page";
+import { HomePage } from "../pages/home.page";
+import { AccountPage } from "../pages/account.page";
+import { ProductPage } from "../pages/product.page";
+import { CheckoutPage } from "../pages/checkout.page";
 
 test("Verify login with valid credentials", async ({ page }) => {
-  await page.goto('/auth/login');
-  await page.getByTestId('email').fill(process.env.USER_EMAIL);
-  await page.getByTestId('password').fill(process.env.USER_PASSWORD);
-  await page.getByTestId('login-submit').click();
-  await expect(page).toHaveURL('/account');
-  await expect(page.getByTestId('page-title')).toContainText("My account");
-  await expect(page.getByTestId('nav-menu')).toContainText(process.env.USER_NAME);
-});
+  const login = new LoginPage(page);
+  const account = new AccountPage(page);
 
+  await login.open();
+  await login.doLogin();
+  await expect(page).toHaveURL(account.path);
+  await expect(account.title).toContainText("My account");
+  await expect(account.navMenu).toContainText(process.env.USER_NAME);
+});
 
 test("Verify user can view product details", async ({ page }) => {
-  await page.goto('/')
-  await page.getByText('Combination Pliers').click();
-  expect(page.url()).toContain('/product');
-  await expect(page.getByTestId('product-name')).toContainText('Combination Pliers');
-  await expect(page.getByTestId('unit-price')).toContainText('14.15');
-  await expect(page.getByTestId('add-to-cart')).toBeVisible();
-  await expect(page.getByTestId('add-to-favorites')).toBeVisible();
+  const home = new HomePage(page);
+  const product = new ProductPage(page);
+  const productName = "Combination Pliers";
+
+  await home.open();
+  await home.openProduct(productName);
+  expect(page.url()).toContain(product.path);
```

Comment:
I'd use the https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-url method

```suggestion
  await expect(page).toHaveURL(product.path);
```
or using reg exp
```suggestion
  await expect(page).toHaveURL(new RegExp(`${product.path}`));
```